modules/service/src/test/scala/lucuma/odb/graphql/query/observingModeGroup.scala

```diff
@@ -0,0 +1,357 @@
+                          }
+                        },
+                        "observations" : {
+                          "matches" : ${g3.map { id => Json.obj("id" -> id.asJson) }.asJson }
+                        }
+                      }
+                    ]
+                  }
+                }
+              """.asRight
+            )
+
+  test("default bin value matches explicit bin value"):
+    createProgramAs(pi).flatMap: pid =>
+      createTargetAs(pi, pid, "Biff",
+         """
+            sourceProfile: {
+              gaussian: {
+                fwhm: {
+                  microarcseconds: 647200
+                }
+                spectralDefinition: {
+                  bandNormalized: {
+                    sed: {
+                      stellarLibrary: B5_III
+                    }
+                    brightnesses: []
+                  }
+                }
+              }
+            }
+          """
+      ).flatMap: tid =>
+
+        def createObs(bin: Option[GmosXBinning]): IO[Observation.Id] =
+          createObservation(
+              pi,
+              pid,
+              Site.GN,
+              grating  = GmosNorthGrating.B1200_G5301.tag.toScreamingSnakeCase,
+              fpu      = GmosNorthFpu.LongSlit_5_00.tag.toScreamingSnakeCase,
+              xbin     = bin,
+              iq       = ImageQuality.PointOne,
+              asterism = List(tid)
+          )
+
+        // Default xbin in first obs, explicit in second obs
+        List(createObs(none), createObs(GmosXBinning.Two.some)).sequence.flatMap: g =>
+          expect(
+            user = pi,
+            query = s"""
+              query {
+                observingModeGroup(programId: ${pid.asJson}) {
+                  matches {
+                    observingMode {
+                      gmosNorthLongSlit {
+                        grating
+                        xBin
+                      }
+                    }
+                    observations {
+                      matches {
+                        id
+                      }
+                    }
+                  }
+                }
+              }
+            """,
+            expected =
+              json"""
+                {
+                  "observingModeGroup" : {
+                    "matches" : [
+                      {
+                        "observingMode" : {
+                          "gmosNorthLongSlit": {
+                            "grating": "B1200_G5301",
+                            "xBin": "TWO"
+                          }
+                        },
+                        "observations" : {
+                          "matches" : ${g.map { id => Json.obj("id" -> id.asJson) }.asJson }
+                        }
+                      }
+                    ]
+                  }
+                }
+              """.asRight
+            )
+
+  test("distinct observing modes edited to become same".ignore):
```

Comment:
This test fails because there are multiple programs with observations using the same value for `c_mode_key`.  The `observations` that are included in the group span different programs despite the filter that limits them to a single program.